### ISSUE TITLE
How to launch integration tests which need twitter credentials when deploying ?

### ISSUE THREAD

**Issue by redouane59** [redouane59/twittered#138]
Currently when deploying only the unit tests are performed but there are no integration tests. Tests in classes like [ITwitterClientV2Test](https://github.com/redouane59/twittered/blob/develop/src/test/java/com/github/redouane59/twitter/nrt/ITwitterClientV2Test.java ) are all disabled because it requieres twitter account credentials to be working fine...

How can we manage this ? Does TravisCI can host securely some credentials that we could use for integration tests ?

**Comment by RakSrinaNa** [comment]
You can encrypt en env variable and use it in the tests.

https://docs.travis-ci.com/user/environment-variables/#defining-encrypted-variables-in-travisyml

**Comment by redouane59** [comment]
> You can encrypt en env variable and use it in the tests.

But should we always use the same twitter account whatever the build or should each developer run the integration tests with his own credentials ?

**Comment by RakSrinaNa** [comment]
The credentials would be bound to this travis repo. So if anyone else run the tests on travis of their own, it'd fail I think.
Though it's probably better that way. So you have control of what the credentials are used for.

Now a thing to take into consideration, is that currently the travis tests are ran automatically on a PR. This means that I could write a test that for example post illicit content and have it run with your credentials.
Would be safer IMO to deactivate automatic build on PR for the tests that use credentials (dunno if possible with travis, it is with github actions tho).

**Comment by redouane59** [comment]
Yes in term of security, it's absolutely not safe if credentials are shared... 
But don't you think that a way exists to store somewhere the developer credentials in order to run automatically integration tests with his own account and be 100% sure that his PR is OK ? 
The target is to decrease a lot possible regressions with external contributions, without any manual step.

**Comment by RakSrinaNa** [comment]
When people will fork the project and have a Travis, it'll be run on there
with their account. They just have to enter their credentials in the env
page.

When a PR is made, it'll be forced and will use your credentials.

**Comment by redouane59** [comment]
> When a PR is made, it'll be forced and will use your credentials.
I don't really like the idea to have to force that. It's a pity that nothing automatic can be implemented :/
